### PR TITLE
MINOR: S3 Add support for single bucket ingestion

### PR DESCRIPTION
**Pull request by ulixius9**
### Describe your changes:

Fixes <issue-number>

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-spec/src/main/resources/json/schema/entity/services/connections/storage/s3Connection.json [hunk 1]
```diff
@@ -24,6 +24,12 @@
       "title": "AWS Credentials Configuration",
       "$ref": "../../../../security/credentials/awsCredentials.json"
     },
+    "bucketName": {
```
Comment (TeddyCr) [inline openmetadata-spec/src/main/resources/json/schema/entity/services/connections/storage/s3Connection.json:27]: Would it make sense to support `n` number of buckets?

Comment (ulixius9) [reply openmetadata-spec/src/main/resources/json/schema/entity/services/connections/storage/s3Connection.json:27]: technically that is supported if you leave this field empty and put filters this is what we follow even for databases right 🤔

Comment (TeddyCr) [reply openmetadata-spec/src/main/resources/json/schema/entity/services/connections/storage/s3Connection.json:27]: I see; what would be the difference with setting a bucketName vs adding a bucket name to the filter?

Comment (ulixius9) [reply openmetadata-spec/src/main/resources/json/schema/entity/services/connections/storage/s3Connection.json:27]: so when you want to ingest multiple buckets you need `ListBuckets` permission, this pr is to cover the scenario when user do not want to give permission for `ListBuckets` due to some security reasons and only wants to ingest a single bucket.


so when using a filter u need `ListBuckets` permissions, when using `bucketName` you do not need that permission

File: ingestion/src/metadata/ingestion/source/storage/s3/connection.py [hunk 2]
```diff
@@ -59,8 +59,15 @@ def test_connection(
     of a metadata workflow or during an Automation Workflow
     """
 
+    def test_buckets(connection: S3Connection, client: S3ObjectStoreClient):
+        if connection.bucketNames:
+            return partial(
+                client.s3_client.list_objects, Bucket=connection.bucketNames[0]
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/storage/s3/connection.py:65]: does it mean we'll only test the connection for the 1st bucket in the list?

Comment (ulixius9) [reply ingestion/src/metadata/ingestion/source/storage/s3/connection.py:65]: yes

**Review by pmbrull — APPROVED** [review]

**Pull request merged**